src/main/java/com/transferwise/openbanking/client/api/payment/v3/RestPaymentClient.java

```
@@ -236,13 +246,21 @@ private void validateResponse(OBWriteDomesticResponse5 response) {
             response.getData().getStatus() == null ||
             response.getData().getDomesticPaymentId() == null ||
             response.getData().getDomesticPaymentId().isBlank()) {
-            throw new ApiCallException("Empty or partial domestic payment response returned " + response);
+            throw new PaymentApiCallException("Empty or partial domestic payment response returned " + response);
         }
     }
 
     private void validateResponse(OBWriteFundsConfirmationResponse1 response) {
         if (response == null || response.getData() == null) {
-            throw new ApiCallException("Empty or partial funds confirmation response returned " + response);
+            throw new PaymentApiCallException("Empty or partial funds confirmation response returned " + response);
+        }
+    }
+
+    private OBErrorResponse1 mapBodyToObErrorResponse(String responseBodyAsString) {
+        try {
+            return jsonConverter.readValue(responseBodyAsString, OBErrorResponse1.class);
+        } catch (Exception ex) {
+            return null;
```

null is good enough for you here? won't it swallow the exception?